app/services/conversation_ai.rb

```diff
@@ -10,10 +7,11 @@ def initialize(user, note)
     @note = note
     @chat = note.chat
     @client = OpenAI::Client.new(access_token: @user.openai_key)
+    @token_handler = TokenHandler.new(@chat)
   end
```

Comment:
The original class I had for OpenAI stuff was not great, but at least it was nicely encapsulated and had no coupling to any specific model. For example, we could use it for a different purpose other than creating a reply.

It's not a good practice to be passing in models like this. The thing to optimize for is: (1) add all methods that pertain to the model, into the actual model itself. There's no reason to shy away from too many methods. You can just put them in a concern if there are a bunch of related methods, but don't create a new class for them. Or (2) stand-alone models that aren't backed by activerecord but have some clear mental container where we definitely need to instantiate something, manipulate it, and hide away a bunch of complexity — but it has nothing to do with the database.